lib/SIL/SILType.cpp

```diff
@@ -555,3 +555,29 @@ SILType SILType::getReferentType(SILModule &M) const {
       getSwiftRValueType()->castTo<ReferenceStorageType>();
   return M.Types.getLoweredType(Ty->getReferentType()->getCanonicalType());
 }
+
+CanType
+SILBoxType::getFieldLoweredType(SILModule &M, unsigned index) const {
+  auto fieldTy = getLayout()->getFields()[index].getLoweredType();
+  
+  // Apply generic arguments if the layout is generic.
+  if (!getGenericArgs().empty()) {
+    // FIXME: Map the field type into the layout's generic context because
+    // SIL TypeLowering currently expects to lower abstract generic parameters
+    // with a generic context pushed, but nested generic contexts are not
+    // supported by TypeLowering. If TypeLowering were properly
+    // de-contextualized and plumbed through the generic signature, this could
+    // be avoided.
```

Comment:
If you want you can take a look at this old PR and see if you can salvage anything: https://github.com/apple/swift/pull/5524/commits